Userland/Libraries/LibPDF/ObjectDerivatives.cpp

```diff
@@ -121,25 +122,40 @@ DeprecatedString DictObject::to_deprecated_string(int indent) const
 DeprecatedString StreamObject::to_deprecated_string(int indent) const
 {
     StringBuilder builder;
+    builder.appendff("{}\n", dict()->to_deprecated_string(indent));
     builder.append("stream\n"sv);
-    append_indent(builder, indent);
-    builder.appendff("{}\n", dict()->to_deprecated_string(indent + 1));
-    append_indent(builder, indent + 1);
 
-    auto string = encode_hex(bytes());
-    while (true) {
-        if (string.length() > 60) {
-            builder.appendff("{}\n", string.substring(0, 60));
-            append_indent(builder, indent);
-            string = string.substring(60);
-            continue;
-        }
+    size_t ascii_count = 0;
+    for (auto c : bytes()) {
+        if (c < 128)
+            ++ascii_count;
+    }
 
-        builder.appendff("{}\n", string);
-        break;
+    size_t percentage_ascii = ascii_count * 100 / bytes().size();
```

Comment:
Is it possible for `bytes()` to be zero-length?

Reply:
Good question. In practice probably not (why encode a stream that has no data), but in theory it can happen. I put in a check for this.